### PR TITLE
[MWPW-136464] CKG search pills revive with enhanced query method

### PR DESCRIPTION
**Pull request by qiyundai**
In this PR we are bringing back search pills for templates pages with more correct way of querying

Resolves: [MWPW-136464](https://jira.corp.adobe.com/browse/MWPW-136464)

Test URLs:
- Before: https://stage--express--adobecom.hlx.page/express/templates/flyer?lighthouse=on
- After: https://mwpw-136464--express--adobecom.hlx.page/express/templates/flyer?lighthouse=on

### CODE REVIEW

File: express/scripts/ckg-link-list.js [hunk 1]
```diff
@@ -137,12 +139,17 @@ async function updateLinkList(container, linkPill, list) {
 
   if (list && templatePages) {
     list.forEach((d) => {
+      const topics = getMetadata('topics') !== '" "' ? `${getMetadata('topics').replace(/[$@%"]/g, '')}` : '';
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:142]: probably should also check if getMetadata wasn't set at all on the page right?

File: express/scripts/ckg-link-list.js [hunk 2]
```diff
@@ -137,12 +139,17 @@ async function updateLinkList(container, linkPill, list) {
...
       const templatePageData = templatePages.find((p) => p.live === 'Y' && matchCKGResult(d, p));
+      const topicsQuery = `${topics ?? topics} ${d.displayValue}`.split(' ')
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:144]: ${topics ?? topics} -> what? :D

File: express/scripts/ckg-link-list.js [hunk 3]
```diff
@@ -137,12 +139,17 @@ async function updateLinkList(container, linkPill, list) {
...
+        .filter((item, i, allItems) => i === allItems.indexOf(item))
+        .join(' ').trim();
       let displayText = formatLinkPillText(d);
 
       const locale = getLocale(window.location);
       const urlPrefix = locale === 'us' ? '' : `/${locale}`;
       const localeColumnString = locale === 'us' ? 'EN' : locale.toUpperCase();
+      let hideUntranslatedPill = false;
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:152]: try to use more positive statements than negative as it's easier to understand. 
I'd change this to `showUntranslatedPill = true`
This also makes your if statement below easier to read `else if (d.ckgID && showUntranslatedPill) {`

File: express/scripts/ckg-link-list.js [hunk 4]
```diff
@@ -153,11 +160,22 @@ async function updateLinkList(container, linkPill, list) {
...
         }
+
+        hideUntranslatedPill = !hasAlternateTextForLocale && locale !== 'us';
       }
 
       if (templatePageData) {
         const clone = replaceLinkPill(linkPill, templatePageData);
         if (clone) pageLinks.push(clone);
+      } else if (d.ckgID && !hideUntranslatedPill) {
+        const currentTasks = getMetadata('tasks') ? getMetadata('tasks').replace(/[$@%"]/g, '') : ' ';
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:171]: stuff like `getMetadata('tasks')` is perfect to put into a variable, so you don't need to call it twice.

File: express/scripts/ckg-link-list.js [hunk 5]
```diff
@@ -104,20 +106,20 @@ async function updateSEOLinkList(container, linkPill, list) {
 }
 
 function formatLinkPillText(linkPillData) {
-  const digestedDisplayValue = titleCase(linkPillData.displayValue.replace(/-/g, ' '));
-  const digestedChildSibling = titleCase(linkPillData.childSibling.replace(/-/g, ' '));
+  const unHyphenedDV = titleCase(linkPillData.displayValue.replace(/-/g, ' '));
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:109]: just call this `DV` :) . unHyphened is too much information to be packing in a variable

File: express/scripts/ckg-link-list.js [hunk 6]
```diff
@@ -104,20 +106,20 @@ async function updateSEOLinkList(container, linkPill, list) {
...
+  const unHyphenedTaskName = titleCase(linkPillData.tasks.replace(/-/g, ' '));
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:110]: same principle here

**Review by vhargrave — COMMENTED** [review]
found a few small things :) . Let me know what you think .

File: express/scripts/ckg-link-list.js [hunk 7]
```diff
@@ -33,14 +33,16 @@ async function fetchLinkList() {
         window.linkLists.ckgData = response.queryResults[0].facets[0].buckets.map((ckgItem) => {
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:33]: this information should not be in the window object btw, but yeah, not part of your task.

File: express/scripts/ckg-link-list.js [hunk 8]
```diff
@@ -104,20 +106,21 @@ async function updateSEOLinkList(container, linkPill, list) {
 }
 
 function formatLinkPillText(linkPillData) {
-  const digestedDisplayValue = titleCase(linkPillData.displayValue.replace(/-/g, ' '));
-  const digestedChildSibling = titleCase(linkPillData.childSibling.replace(/-/g, ' '));
-  const topics = getMetadata('topics') !== '" "' ? `${getMetadata('topics').replace(/[$@%"]/g, '').replace(/-/g, ' ')}` : '';
+  const dv = titleCase(linkPillData.displayValue.replace(/-/g, ' '));
```
Comment (vhargrave) [inline express/scripts/ckg-link-list.js:109]: sorry for not noticing this earlier, using acronyms is also not great. Would be best to call this digestedValue and the lower variable titleName.

**Review by vhargrave — COMMENTED** [review]
looking pretty good!

**Review by vhargrave — APPROVED** [review]

**Review by apganapa — APPROVED** [review]

**Pull request merged**